src/OrleansRuntime/ReminderService/ReminderRegistry.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Orleans.Runtime;
+
+namespace Orleans.Timers
+{
+    internal class ReminderRegistry : IReminderRegistry
+    {
+        private const uint MaxSupportedTimeout = 0xfffffffe;
+
+        private readonly ISiloRuntimeClient runtimeClient;
+        
+        public ReminderRegistry(ISiloRuntimeClient runtimeClient)
+        {
+            this.runtimeClient = runtimeClient;
+        }
+
+        public Task<IGrainReminder> RegisterOrUpdateReminder(string reminderName, TimeSpan dueTime, TimeSpan period)
+        {
```

Comment:
The implementation of this was taken from the removed `ReminderRegistry` + the corresponding removed methods on `InsideRuntimeSilo`. The only real difference is that instead of using the global `RuntimeClient.Current`, we inject the `ISiloRuntimeClient`